src/platform/plugins/shared/controls/public/control_group/components/control_panel.tsx

```diff
@@ -122,49 +97,61 @@ export const ControlPanel = <ApiType extends DefaultControlApi = DefaultControlA
   const controlWidth = width ?? DEFAULT_CONTROL_WIDTH;
   const controlGrow = grow ?? DEFAULT_CONTROL_GROW;
+  const hasRoundedBorders = !api?.CustomPrependComponent && !isEditable && isTwoLine;
+  const shouldHideComponent = Boolean(blockingError);
+
+  const insertBefore = isOver && (index ?? -1) < (activeIndex ?? -1);
+  const insertAfter = isOver && (index ?? -1) > (activeIndex ?? -1);
+
+  const styles = useMemoizedStyles(controlPanelStyles);
+
   return (
     <EuiFlexItem
       ref={setNodeRef}
-      style={style}
+      style={{
+        transition,
+        transform: isSorting ? undefined : CSS.Translate.toString(transform),
+      }}
       grow={controlGrow}
       data-control-id={uuid}
       data-test-subj="control-frame"
       data-render-complete="true"
-      className={classNames('controlFrameWrapper', {
-        'controlFrameWrapper--small': controlWidth === 'small',
-        'controlFrameWrapper--medium': controlWidth === 'medium',
-        'controlFrameWrapper--large': controlWidth === 'large',
-        'controlFrameWrapper-isDragging': isDragging,
-        'controlFrameWrapper--insertBefore': isOver && (index ?? -1) < (activeIndex ?? -1),
-        'controlFrameWrapper--insertAfter': isOver && (index ?? -1) > (activeIndex ?? -1),
-      })}
+      css={css([
+        isDragging && styles.draggingItem,
+        styles.responsiveControlWidthStyles,
+        controlPanelWidthStyles(controlWidth),
+      ])}
     >
       <FloatingActions
+        data-test-subj="control-frame-floating-actions"
         api={api}
-        className={classNames({
-          'controlFrameFloatingActions--twoLine': usingTwoLineLayout,
-          'controlFrameFloatingActions--oneLine': !usingTwoLineLayout,
-        })}
+        isTwoLine={isTwoLine}
         viewMode={viewMode}
         disabledActions={disabledActionIds}
         isEnabled={true}
       >
         <EuiFormRow
           data-test-subj="control-frame-title"
           fullWidth
-          label={usingTwoLineLayout ? panelTitle || defaultPanelTitle || '...' : undefined}
+          label={isTwoLine ? panelTitle || defaultPanelTitle || '...' : undefined}
         >
           <EuiFormControlLayout
             fullWidth
             isLoading={Boolean(dataLoading)}
-            className={classNames(
-              'controlFrame__formControlLayout',
-              {
-                'controlFrame__formControlLayout--twoLine': usingTwoLineLayout,
-                'controlFrame__formControlLayout--edit': isEditable,
-              },
-              `${controlType}`
-            )}
+            className={controlType}
+            css={css([
+              styles.formControl,
+              isEditable
+                ? isTwoLine
+                  ? styles.formControlEditableTwoLines
+                  : styles.formControlEditableOneLine
+                : isTwoLine
+                ? styles.formControlNonEditableTwoLines
+                : styles.formControlNonEditableOneLine,
```

Comment:
We try to avoid nested ternaries like this - see https://docs.elastic.dev/kibana-dev-docs/contributing/styleguide#only-use-ternary-operators-for-small-simple-code